### ISSUE TITLE
Feature Request: Contribute commands with additional arguments

### ISSUE THREAD

**Issue by austenc** [microsoft/vscode#41692]
Hello, I'm creating an extension to provide easily accessible documentation shortcuts from the command palette in the editor. Currently it does not seem possible to do something like this (where `otherArg` would be an arbitrary argument in package.json):
```
    "contributes": {
        "commands": [
            {
                "command": "extension.doSomething",
                "title": "Example Title",
                "otherArg": "some other argument"
            }
        ]
    },
```

I see that a similar question has been asked here https://github.com/Microsoft/vscode/issues/26436 and also in a resulting (unanswered) SO post here https://stackoverflow.com/questions/43909741/can-i-pass-arguments-to-command-in-contributes-block.

Is this a feature that would be considered? If not, how might this type of thing be accomplished from the extension developer's perspective? I see there is an `args` parameter for command handler functions, but I'm not seeing a way to call `executeCommand` from the package.json file. 

For reference, here is a screenshot of how my equivalent package looks in the command palette in sublime text:
![image](https://user-images.githubusercontent.com/575421/35001510-27321b58-faa4-11e7-9f4b-c26b05408730.png)

Thanks!

**Comment by jrieken** [comment]
I believe @bpasero has done that, right?

**Comment by austenc** [comment]
@jrieken Thanks -- interested to see how it might be done. Specifically how one might access additional params from the `registerCommand` handler in the `extension.ts` file.

**Comment by bpasero** [comment]
@jrieken for keybindings, but not for extension commands.

**Comment by austenc** [comment]
@bpasero @jrieken Thank you to both of you for discussing / entertaining this idea. My thought was it would allow extension developers to much more easily reuse commands. In the meantime (as a workaround) is it possible to get the `title` attribute from within the command's handler function?

**Comment by jrieken** [comment]
Yeah, it would probably not be on the command itself then but in the menu section (since the command might be shared across different menus).

**Comment by austenc** [comment]
If you wanted to have a command that simply opens a (variable) url depending on which item was selected in the command palette, wouldn't you still need multiple commands (one for each command palette entry) defined with the menu items approach? The `args` support for keybindings seems like a natural fit for commands too. 

Perhaps I'm missing something, but with the menu section approach there's still no way to pass any sort of args to the command right? I'd rather not have two dozen commands that all do basically the same thing! :)
